msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/AccountState.kt

```diff
@@ -102,63 +105,75 @@ class AccountState private constructor(
+                            MsalClientException.NO_CURRENT_ACCOUNT,
+                            MsalClientException.NO_CURRENT_ACCOUNT_ERROR_MESSAGE
+                        )
+
+                val requestAccountRecord = AccountRecord()
+                requestAccountRecord.environment = (account as Account).environment
+                requestAccountRecord.homeAccountId = account.homeAccountId
+
+                val params = CommandParametersAdapter.createRemoveAccountCommandParameters(
+                    config,
+                    config.oAuth2TokenCache,
+                    requestAccountRecord
+                )
 
-            val result = CommandDispatcher.submitSilentReturningFuture(removeCurrentAccountCommandParameters)
-                .get().result as Boolean
+                val removeCurrentAccountCommandParameters = RemoveCurrentAccountCommand(
+                    params,
+                    LocalMSALController().asControllerFactory(),
+                    object : CommandCallback<Boolean?, BaseException?> {
+                        override fun onError(error: BaseException?) {
+                            // Do nothing, handled by CommandDispatcher.submitSilentReturningFuture()
+                        }
+
+                        override fun onTaskCompleted(result: Boolean?) {
+                            // Do nothing, handled by CommandDispatcher.submitSilentReturningFuture()
+                        }
+
+                        override fun onCancel() {
+                            // Do nothing
+                        }
+                    },
+                    PublicApiId.NATIVE_AUTH_ACCOUNT_SIGN_OUT
+                )
 
-            return@withContext if (result) {
-                SignOutResult.Complete
-            } else {
-                Logger.error(
-                    TAG,
-                    "Unexpected error during signOut.",
-                    null
+                val result = CommandDispatcher.submitSilentReturningFuture(
+                    removeCurrentAccountCommandParameters
                 )
-                throw MsalClientException(
-                    MsalClientException.UNKNOWN_ERROR,
-                    "Unexpected error during signOut."
+                    .get().result as Boolean
+
+                return@withContext if (result) {
+                    SignOutResult.Complete
+                } else {
+                    Logger.error(
+                        TAG,
+                        "Unexpected error during signOut.",
+                        null
+                    )
+                    throw MsalClientException(
+                        MsalClientException.UNKNOWN_ERROR,
+                        "Unexpected error during signOut."
+                    )
+                }
+            } catch (e: Exception) {
+                Logger.error(TAG, "MSAL client exception occurred in getCurrentAccount.", e)
+                ClientExceptionError(
```

Comment:
For sign in, sign up, reset password cases, use their own errors instead (SignInError, SignUpError, ResetPasswordError). for those do not have corresponding errors, keep ClientExceptionError as a placeholder.

Reply:
Sorry, `UnknownError` is not the right class. I mean to say: why not use the existing generic error classes for this case, i.e. `SignInError`, `SignupError`, `ResetPasswordError`. Similar to 
<img width="350" alt="image" src="https://github.com/AzureAD/microsoft-authentication-library-for-android/assets/1315628/a213066c-10e1-43f4-844c-99a5f6825430">

Reply:
For those without the existing generic error classes.  Do we need to create those generic error classes for them?  I personally think it would be better than ClientExceptionError